cpp/log/cert.cc

```diff
@@ -1258,9 +1290,10 @@ Cert::Status CertChain::IsValidCaIssuerChainMaybeLegacyRoot() const {
     // The root cert may not have CA:True
     status = issuer->IsSelfSigned();
     if (status == Cert::FALSE) {
-      Cert::Status s2 = issuer->HasBasicConstraintCATrue();
-      if (s2 != Cert::TRUE)
-        return s2;
+      StatusOr<bool> s2 = issuer->HasBasicConstraintCATrue();
+      if (StatusOrBoolToCertStatus(s2) != Cert::TRUE) {
+        return StatusOrBoolToCertStatus(s2);
```

Comment:
Especially doing it twice in a row, could have done this instead (styled with my beloved `const` and constructor-style initialisation):

```C++
const Cert::Status s2(StatusOrBoolToCertStatus(issuer->HasBasicConstraintCATrue()));
```